rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake

```diff
@@ -17,10 +17,19 @@ find_package(rosidl_runtime_c REQUIRED)
 find_package(rosidl_typesupport_c REQUIRED)
 find_package(rosidl_typesupport_interface REQUIRED)
 
-find_package(PythonInterp 3.6 REQUIRED)
-
-find_package(python_cmake_module REQUIRED)
-find_package(PythonExtra MODULE REQUIRED)
+find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
+if(WIN32 AND CMAKE_BUILD_TYPE STREQUAL "Debug")
```

Comment:
IIUC it looks like this is using the path to the debug executable as a hint to find the debug version of Numpy, but in doing so it duplicates the logic in PythonExtra to find the debug version of Python.

One option, I see https://github.com/ros2/python_cmake_module/pull/7 references a `PythonExtra_EXECUTABLE` variable. How about we promote that to an output variable so that code here could set the `Python3_EXECUTABLE` hint to it's value?

Another option, what if `PythonExtra` supported components, such that it passed the given components on to its own `find_package(Python3 ...)` call?

Reply:
I did something slightly different.  I ended up reinstating the `find_package` calls to both `python_cmake_module` and `PythonExtra`, and then fetching the target property `Python3_EXECUTABLE` from `PythonExtra::Interpreter`.  I believe that should have the desired effect, while still reusing the logic from `python_cmake_module`.